app/models/product_drive.rb

```diff
@@ -76,15 +77,29 @@ def item_quantities_by_name_and_date(date_range)
     end
   end
 
-  def donation_quantity_by_date(date_range)
-    donations.joins(:line_items)
-      .during(date_range)
-      .sum('line_items.quantity')
+  def donation_quantity_by_date(date_range, item_category_id = nil)
+    if [nil, ''].include? item_category_id
```

Comment:
More Rails-y: `if item_category_id.blank?` Better yet, use `present?` and switch the conditional so the positive condition comes first.